query/esclauses.go

```diff
@@ -0,0 +1,77 @@
+package query
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	Separator = ","
+
+	NoProvisionalNoConfirmedPostponed = `
+{"term":{"description.finalised":true}}, {"exists":{"field":"dateChanges"}}
+`
+
+	NoProvisionalConfirmedNoPostponed = `
+{"term":{"description.finalised":true}}, {"bool":{"must_not":{"exists":{"field":"dateChanges"}}}}
+`
+
+	NoProvisionalConfirmedPostponed = `
+{"term":{"description.finalised":true}}
+`
+
+	ProvisionalNoConfirmedNoPostponed = `
+{"term":{"description.finalised":false}}
+`
+
+	ProvisionalNoConfirmedPostponed = `
+{"bool":{
+    "should":[
+      {"term":{"description.finalised":false}},
+      {"bool":{
+          "must":[
+            {"term":{"description.finalised":true}},
+            {"exists":{"field":"dateChanges"}}
+          ]}
+      }
+    ]}
+}`
+
+	ProvisionalConfirmedNoPostponed = `
+{"bool":{
+    "should":[
+      {"term":{"description.finalised":false}},
+      {"bool":{
+          "must":[
+            {"term":{"description.finalised":true}},
+            {"bool":{"must_not":{"exists":{"field":"dateChanges"}}}}
+          ]}
+      }
+    ]}
+}`
+)
+
+func mainUpcomingClause(now time.Time) string {
+	return fmt.Sprintf("%s, %s, %s", `{"term": {"description.published": false}}`, `{"term": {"description.cancelled": false}}`,
+		fmt.Sprintf(`{"range": {"description.release_date": {"gte": %q}}}`, now.Format(dateFormat)))
+}
+
+//gocyclo:ignore
```

Comment:
(question): why do we have to ignore gocylco here , is there no way to remove the cyclic dependency ?

Reply:
Yes the function has a cyclomatic complexity greater than the limit, but this is because of the switch statement which is exactly what the function was written for, i.e. expressing a boolean matrix of the 3 boolean values for Provisional, Confirmed and Postponed. There is really no way around it

Reply:
sure, one last question I guess you can increase the gocyclo complexity in golangcilint.yaml and in that way this operates on a higher level of complexity rather than completely ignoring it. What do you think ? 

```
 gocyclo:
    min-complexity: 40
    ```